src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt

```diff
@@ -69,9 +92,10 @@ class CosmasGoogleCloudService(private val bucketName: String,
 
     override fun createPatch(request: CosmasProto.CreatePatchRequest,
                              responseObserver: StreamObserver<CosmasProto.CreatePatchResponse>) {
-        LOG.info("Get request for create new patch of file # ${request.fileId} by user ${request.patchesList[0].userId}")
+        LOG.info("Get request for create new patch of file # ${request.fileId} " +
```

Comment:
Remove string concatenation? Also, just if it happens that `request.patchesList` is empty, this will fail in the logging attempt. Also, despite that Kotlin makes string interpolation so easy, it is recommended to pass pattern and arguments to loggers, especially if they are called frequently. The matter is that string interpolation is not free, and message constructed every time, no matter if the message is logged or not, while pattern+arguments will turn into the final message only if it is going to be logged.

Reply:
done